### DISCUSSION TITLE
[macos] "Open folder with wezterm" from finder context menu

### DISCUSSION THREAD

**Discussion by ghry5** [wezterm/wezterm#4627] [Show and tell]
Was missing the simple functionality to have an option in the context menu to open a folder directly in wezterm from finder. Its actually super easy to set up:

1. Open Automator: Search for Automator in Spotlight and open it 
2. Create new Quick Action
3. Set `Worflow receives current` option to "folders" in "Finder.app", set any custom Image.
4. On the left side search for "Run shell script"
5. Change `Pass input` to "as arguments"
6. add the following script: 
    ```
    for f in "$@"
    do
	/opt/homebrew/bin/wezterm start --cwd "$f"
    done
    ```
7. Hit File > Save and give it a name: "Open in Wezterm"

Now, when you right-click a folder in Finder, you should see the "Open in Wezterm" option in the context menu under Quick Actions > Open in Wezterm. Clicking it will open the selected folder in Wezterm.

I also added a "open file/folder in neovim (in wezterm)" by adjusting the quick action like requested here: https://github.com/wez/wezterm/issues/3990

**Comment by windoze** [comment]
Seems this quick action doesn't work with symbolic links to directories, the action doesn't appear in the "Quick Actions" menu.
Any idea?

**Reply by ian-h-chamberlain** [reply]
This seems to work for me, make sure you've set the workflow to receive "files or folders" (since a symlink is a file, not a directory):

<img width="535" alt="Screenshot 2025-04-22 at 07 57 14" src="https://github.com/user-attachments/assets/801fd2ad-d99d-45ce-bc9a-1497defd516f" />

```sh
for f in "$@"
do
	if [[ -L "$f" ]]; then
		f=$(readlink -f "$f")
	fi
	
	if [[ -d "$f" ]]; then 
		/opt/homebrew/bin/wezterm start --cwd "$f"
	else
		# start in the parent dir instead
		/opt/homebrew/bin/wezterm start --cwd "$(dirname "$f")"
	fi
done
```
